nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py

```diff
@@ -230,7 +231,16 @@ def setup_experiment(self, config):
 
         # Configure model
         self.device = config.get("device", self.device)
-        self.model = self.create_model(config, self.device)
+        self.model = self.create_model(config, from_checkpoint=False)
+
+        checkpoint_file = config.get("checkpoint_file", None)
+        if checkpoint_file:
+            load_checkpoint_args = config.get("load_checkpoint_args", {})
+            self.restore_checkpoint(self.model, checkpoint_file, load_checkpoint_args)
+
+        # Send to device after loading checkpoint
+        self.transform_model()
```

Comment:
This is forcing us to always restore checkpoints on CPU rather than on CUDA, then to move it back to CUDA (even in our main experiments when we're not using QAT yet). Maybe instead, the QuantizationAware mixin should override create_model and put all of this logic there. I'm not sure we need `transform_model` and `restore_checkpoint`, it seems the current ImagenetExperiment is already sufficient to implement this.

Then you could use the `device=None` option in `network_utils.create_model`, but you won't need the `from_checkpoint` bool any more.

Reply:
Do you see a big impact in performance? I understand your point, but the added flexibility seemed like a good trade compared to at most a few seconds more only in initialization, but I might have estimated it wrongly.

Reply:
And functionally, from a SRP principle, I think restore checkpoints and creating models should be separate methods.

Reply:
The last commit should address your concerns. There is still a transform model, but create_model is unchanged, so shouldn't impact existing experiments. I had to a little bit of hacking in the `quantize_qat` script to account for the different order of checkpoint restoration - in a later refactor we should disentangle restore_checkpoint from create model.